lib/system.nim

```diff
@@ -2722,12 +2722,26 @@ proc procCall*(x: untyped) {.magic: "ProcCall", compileTime.} =
   ##   ```
   discard
 
+proc strcmp(a, b: cstring): cint {.noSideEffect,
+  importc, header: "<string.h>".}
+
+proc `<`*(x, y: cstring): bool {.magic: "LtCString", noSideEffect,
+                                   inline.} =
+  if pointer(x) == nil or pointer(y) == nil:
```

Comment:
this is odd (because `cstring(nil) < ""` and `"" < cstring(nil)` are both true at the same time

Reply:
(while `cstring(nil) < cstring(nil)` remains false)

Reply:
It seems that JS agrees with my order

```js
console.log((null < ""));
console.log(("" < null));
console.log(("" == null));
console.log((null == null));
```

```
false
false
false
true
```

Reply:
It seems that JS agrees on my order

```js
import std/jsconsole

console.log cstring(nil) < cstring""
console.log cstring"" < cstring(nil)
console.log cstring(nil) > cstring""
console.log cstring"" > cstring(nil)
console.log cstring"" == cstring(nil)
console.log cstring(nil) == cstring(nil)
```

```
false
false
false
false
false
true
```
I'm not sure whether it's worth handling NUL: differently

Reply:
I don't think js is the model to follow here since js does type coersion and other tricks (ie in javascript, you can do `0 < "1"` and expect `true`. What javascript does in that case is essentially that it returns `false` for any "undefined" comparison.

By placing `null` before `""`, we can meaningfully work with it in algorithms like `lowerBound` etc by _defining_ a behavior that is logically consistent - otherwise, such utilities are undefined as well - at that point, it would be better to disallow `null` completely.

Reply:
Okay